backend/src/main/java/com/capstone/moneytree/service/impl/DefaultUserService.java

```diff
@@ -258,31 +264,75 @@ public User login(User credentials)
    }
 
    @Override
-   public Long followUser(Long userId, Long userToFollowId) {
+   public String followUser(Long userId, Long userToFollowId) {
+      if (userId == userToFollowId) {
+         throw new FollowsRelationshipException(USER_CANT_BE_FOLLOWED_BY_ITSELF);
+      }
       User user = userDao.findUserById(userId);
       User userToFollow = userDao.findUserById(userToFollowId);
       if (user == null || userToFollow == null) {
          throw new EntityNotFoundException(USER_NOT_FOUND);
       }
+      List<Follows> allFollowRels = followsDao.findFollowsByFollowerId(userId);
+      // check to see if user already following the other user
+      for (Follows rel : allFollowRels) {
+         if (rel.getUserToFollow().getId() == userToFollowId) {
+            throw new FollowsRelationshipException(USER_ALREADY_FOLLOWED);
+         }
+      }
+      Date currentDate = new Date();
+      Follows newFollowRel = new Follows(user, userToFollow, currentDate);
+      followsDao.save(newFollowRel);
 
-      user.follow(userToFollow);
-      userDao.save(user);
-
-      return userToFollowId;
+      return user.getUsername() + " followed " + userToFollow.getUsername();
```

Comment:
Ahh I see now what we're giving to the frontend as a response. Maybe we could create a FollowResponse object containing the `user.getUsername()` and `userToFollow.getUsername()` inside? An object as such:
```
{
username: 'string'
follows: 'string'
}

That way, if we ever change anything on the frontend, then we don't have to change the backend as well. I feel like returning a string is prone to changes since we may have to edit the string format. Having an object gives the frontend a way to do whatever they want with the data. What do you think?
```

Reply:
Actually for the frontend part we only care about response code to be 2XX so I think it would be better to revert it to Long id.

Reply:
Yes. Read the comment up there about that.